From 8f8e12968f23ee7f45e01c2cf84e29d15a3e125d Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 25 Jul 2017 15:24:02 +0800
Subject: [PATCH 2204/5242] MLK-16075-9 gpu: imx: dpu: common: Add helper to
 get number of planes of format

commit  0347ba7956013b3cedfb807981cf2874d4892092 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds a helper dpu_format_num_planes() to get number of planes
of pixel formats which are supported by the current dpu base driver.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |   15 +++++++++++++++
 drivers/gpu/imx/dpu/dpu-prv.h    |    1 +
 2 files changed, 16 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index dd468d5..2f280b7 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -575,6 +575,21 @@ int dpu_format_vert_chroma_subsampling(u32 format)
 	}
 }
 
+int dpu_format_num_planes(u32 format)
+{
+	switch (format) {
+	case DRM_FORMAT_NV12:
+	case DRM_FORMAT_NV21:
+	case DRM_FORMAT_NV16:
+	case DRM_FORMAT_NV61:
+	case DRM_FORMAT_NV24:
+	case DRM_FORMAT_NV42:
+		return 2;
+	default:
+		return 1;
+	}
+}
+
 #define DPU_UNIT_INIT(dpu, base, unit, name, id, pec_ofs, ofs)		\
 {									\
 	int ret;							\
diff --git a/drivers/gpu/imx/dpu/dpu-prv.h b/drivers/gpu/imx/dpu/dpu-prv.h
index 25066cf..f8e6e7f 100644
--- a/drivers/gpu/imx/dpu/dpu-prv.h
+++ b/drivers/gpu/imx/dpu/dpu-prv.h
@@ -228,6 +228,7 @@ struct dpu_soc {
 
 int dpu_format_horz_chroma_subsampling(u32 format);
 int dpu_format_vert_chroma_subsampling(u32 format);
+int dpu_format_num_planes(u32 format);
 
 #define DECLARE_DPU_UNIT_INIT_FUNC(block)			\
 int dpu_##block##_init(struct dpu_soc *dpu, unsigned int id,	\
-- 
1.7.9.5

